source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,207 @@
+        {
+            for (size_t j = 0; j < pointCloud.width; j++)
+            {
+                if (std::isnan(pointCloud(i, j).z))
+                {
+                    auto& zRGB = zJetColorMap.at<cv::Vec3b>(i, j);
+                    zRGB[0] = 0;
+                    zRGB[1] = 0;
+                    zRGB[2] = 0;
+                }
+            }
+        }
+        return zJetColorMap;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGB> maskPointCloud(const Zivid::PointCloud& pointCloud, const cv::Mat& mask) 
+    {
+        const auto data = pointCloud.copyPointsXYZColorsRGBA();
+        const int height = data.height();
+        const int width = data.width();
+        const float bad_point = std::numeric_limits<float>::quiet_NaN();
+
+        // Creating point cloud structure
+        pcl::PointCloud<pcl::PointXYZRGB> maskedPointCloud(height, width);
+        maskedPointCloud.is_dense = false;
+        maskedPointCloud.points.resize(height * width);
+
+        std::cout << "Masking point cloud" << std::endl;
+
+        // Copying data points within the mask. Rest is set NaN
+        for (int i = 0; i < height; i++)
+        {
+            for (int j = 0; j < width; j++)
+            {
+                if (mask.at<uint8_t>(i, j) > 0) 
+                {
+                    maskedPointCloud(i, j).r = data(i, j).color.r;
+                    maskedPointCloud(i, j).g = data(i, j).color.g;
+                    maskedPointCloud(i, j).b = data(i, j).color.b;
+                    maskedPointCloud(i, j).x = data(i, j).point.x;
+                    maskedPointCloud(i, j).y = data(i, j).point.y;
+                    maskedPointCloud(i, j).z = data(i, j).point.z;
+                }
+                else 
+                {
+                    maskedPointCloud(i, j).r = bad_point;
+                    maskedPointCloud(i, j).g = bad_point;
+                    maskedPointCloud(i, j).b = bad_point;
+                    maskedPointCloud(i, j).x = bad_point;
+                    maskedPointCloud(i, j).y = bad_point;
+                    maskedPointCloud(i, j).z = bad_point;
+                }
+            }
+        }
+        return maskedPointCloud;
+    }
+
+    void visualizeDepthMap(const pcl::PointCloud<pcl::PointXYZRGB>& pointCloud) 
+    {
+        std::cout << "Converting to Depth map in OpenCV format" << std::endl;
+        cv::Mat zJetColorMap = pointCloudToCvZ(pointCloud);
+        std::cout << "Visualizing Depth map" << std::endl;
+        cv::namedWindow("Depth map", cv::WINDOW_AUTOSIZE);
+        cv::imshow("Depth map", zJetColorMap);
+        cv::waitKey(0);
+    }
+
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        // Reading ZDF-file
+        std::string ZDFFileName = std::string(ZIVID_SAMPLE_DATA_DIR) + "/MiscObjects.zdf";
+        std::cout << "Reading ZDF frame from file: " << ZDFFileName << std::endl;
+        const auto frame = Zivid::Frame(ZDFFileName);
```

Comment:
We try to use some standardization between the samples. This is related to description, comments, print outs, and variable names.

Check and implement this (fileName instead of ZDFFileName): https://github.com/zivid/zivid-cpp-samples/blob/a97ca886ec9dca02b18cca02811776b72058798c/source/Applications/Advanced/Downsample/Downsample.cpp#L178-L180